app/forms/candidate_interface/degree_wizard.rb

```diff
@@ -461,6 +420,75 @@ def specified_grades?
 
   private
 
+    def reviewing_or_country_not_changed_steps(step)
+      case step
+      when :degree_level
+        degree_has_type? ? :type : :review
+      when :completed
+        :award_year
+      when :award_year
+        completed? && international? ? :enic : :review
+      when :enic
+        if international?
+          enic_reason == HAS_STATEMENT ? :enic_reference : :review
+        else
+          :review
+        end
+      else
+        :review
+      end
+    end
+
+    def uk_steps(step)
+      return :type if step == :subject && degree_has_type?
+      return :start_year if step == :completed && phd?
+
+      next_step = {
+        country: :degree_level,
+        degree_level: :subject,
+        subject: :university,
+        type: :university,
+        university: :completed,
+        completed: :grade,
+        grade: :start_year,
+        start_year: :award_year,
+        award_year: :review,
+      }
+
+      next_step[step] || handle_invalid_step
```

Comment:
Can we do this as a switch statement to be consistent with the pattern above?

Reply:
> I think it would be helpful to be as consistent as possible with our patterns within the class. I felt like I was switching a lot between early returns, ternaries, switches, if/else.

I understand. That makes sense and I do generally agree with you. However in this case, there is no good way to do this. Again this points to the degree wizard needing rewrite rather than a refactor. 

I have refactored this to a case statement [here](https://github.com/DFE-Digital/apply-for-teacher-training/pull/10142/commits/b4adc79c822db6353555dd91c68cdc386aab3651) for demonstration purposes, but i'm not convinced it's better for a couple of reasons: 

- We need two case statements to handle early returns and that may not be obvious to everyone when updating the code, meaning it's less maintainable. It's more obvious what's happening when we use `return` at the top of the method.

- Using a hash for the straightforward cases makes the code concise and easier to extend when/if we need to add to it.

Reply:
I agree that the degree wizard needs a rewrite. Essentially, it is doing the work of about three wizards. I suppose it's a question of whether moving things around in this refactor makes things more or less easy to follow.